user_tools/src/spark_rapids_pytools/rapids/qualification.py

```diff
@@ -44,97 +44,91 @@ class QualificationSummary:
     """
     Encapsulates the logic to organize Qualification report.
     """
-    comments: Any = None
-    all_apps: pd.DataFrame = None
-    recommended_apps: pd.DataFrame = None
-    df_result: pd.DataFrame = None
+    total_apps: pd.DataFrame = field(init=True)  # Total apps, including failed or skipped
+    tools_processed_apps: pd.DataFrame = None  # Apps after tools processing and heuristic filtering
+    recommended_apps: pd.DataFrame = None  # Apps recommended by legacy speedups. TODO: Should use QualX
+    filter_apps_count: int = field(default=0, init=False)  # Count after applying console filters (top candidates)
     top_candidates_flag: bool = False
+    comments: Any = None
     sections_generators: List[Callable] = field(default_factory=lambda: [])
-    filter_apps_count: int = field(default=0, init=False)
-    conversion_items: dict = field(default_factory=dict)
-    auto_tuning_path: str = None
 
     def _get_total_durations(self) -> int:
-        if not self.is_empty():
-            return self.all_apps['App Duration'].sum()
+        if self._has_tools_processed_apps():
+            return self.tools_processed_apps['App Duration'].sum()
         return 0
 
     def _get_total_gpu_durations(self) -> int:
-        if not self.is_empty():
-            return self.all_apps['Estimated GPU Duration'].sum()
+        if self._has_tools_processed_apps():
+            return self.tools_processed_apps['Estimated GPU Duration'].sum()
         return 0
 
     def _get_stats_total_cost(self) -> float:
-        return self.df_result['Estimated App Cost'].sum()
+        if self._has_tools_processed_apps() and 'Estimated App Cost' in self.tools_processed_apps.columns:
+            return self.tools_processed_apps['Estimated App Cost'].sum()
+        return 0.0
 
     def _get_stats_total_gpu_cost(self) -> float:
-        return self.df_result['Estimated GPU Cost'].sum()
+        if self._has_tools_processed_apps() and 'Estimated GPU Cost' in self.tools_processed_apps.columns:
```

Comment:
same as above

Reply:
Removal of dead code related to cost-savings might be be done in #1221